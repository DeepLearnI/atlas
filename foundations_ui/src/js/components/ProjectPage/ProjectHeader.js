import React from 'react';
import PropTypes from 'prop-types';

const ProjectHeader = props => {
  const { numProjects } = props;

  return (
    <div className="project-header-container">
      <div className="project-header-logo-container">
        <div className="i--icon-logo" />
        <h2 className="font-bold">Foundations</h2>
      </div>
      <div className="project-header-info-container">
        <div className="project-header-total-projects-container">
          <div className="half-width inline-block">
            <h1 className="blue-border-bottom font-bold">Projects</h1>
          </div>
          <div className="half-width inline-block text-right">
            <h2 className="blue-border-bottom font-bold">
              Total Projects: <span>{numProjects}</span>
            </h2>
          </div>
        </div>
        <div className="project-header-sort-filter-container" />
      </div>
    </div>
  );
};

ProjectHeader.propTypes = {
  numProjects: PropTypes.number,
};

ProjectHeader.defaultProps = {
  numProjects: 0,
};

export default ProjectHeader;
