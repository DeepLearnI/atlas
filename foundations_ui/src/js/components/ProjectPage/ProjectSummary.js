import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import HoverCell from '../JobListPage/cells/HoverCell';
import CommonActions from '../../actions/CommonActions';

const ProjectSummary = props => {
  const [showAllTags, setShowAllTags] = React.useState(false);

  const packageClick = () => {
    const { history, project } = props;

    history.push(
      `/projects/${project.name}/job_listing`,
      {
        project: project,
      },
    );
  };

  const onClickShowTags = () => {
    const { project } = props;
    const value = !showAllTags;
    setShowAllTags(value);
  };

  const onMouseLeave = () => {
    setShowAllTags(false);
  };

  const { project } = props;

  const expandedTagSpans = [];
  project.tags.forEach(tag => {
    expandedTagSpans.push(<span key={'tag-'.concat(tag)}>{tag}</span>);
  });

  return (
    <div
      className="project-summary-container"
      data-class="project-summary"
    >
      <div className="project-summary-info-container">
        <h2
          onClick={packageClick}
          // eslint-disable-next-line jsx-a11y/no-noninteractive-element-to-interactive-role
          role="button"
          tabIndex={0}
          onKeyPress={packageClick}
          className="font-bold"
        >
          {project.name}
        </h2>
        <p className="font-bold">
          Created at: <span>{CommonActions.formatDate(project.created_at)}</span>
        </p>
        <div className="project-summary-button-container" />
      </div>
      <div className="project-summary-tags-container">
        <p>tags</p>
        {project.tags.slice(0, 10).map(tag => {
          return <span key={tag}>{tag}</span>;
        })}
        {project.tags.length > 10
          && (
            <span
              className="span-more"
              role="presentation"
              onClick={onClickShowTags}
              onKeyDown={() => {}}
            >
              ...
            </span>
          )
        }
        {showAllTags && <HoverCell onMouseLeave={onMouseLeave} textToRender={expandedTagSpans} />}
      </div>
    </div>
  );
};

ProjectSummary.propTypes = {
  project: PropTypes.object,
  history: PropTypes.object,
};

ProjectSummary.defaultProps = {
  project: {},
  history: {},
};

export default withRouter(ProjectSummary);
