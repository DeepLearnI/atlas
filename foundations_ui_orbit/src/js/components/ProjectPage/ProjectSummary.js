import React from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";

const ProjectSummary = props => {
  const packageClick = () => {
    const { history, project } = props;

    history.push(
      `/projects/${project.name}/schedules`,
      {
        project: project
      }
    );
  };

  const { project } = props;


  return (
    <div
      className="project-summary-container elevation-1"
      onClick={packageClick}
    >
      <div className="project-summary-info-container">
        <h2 className="font-bold">{project.name}</h2>
        <p>Data Source: Unknown</p>
        <p className="font-bold">
          Project owner: <span>{project.owner}</span>
        </p>
        <p className="font-bold">
          Created at: <span>{project.created_at}</span>
        </p>
        <div className="project-summary-button-container">
          {/* <button type="button" className="b--mat b--affirmative">
              view queue
            </button>
            <button type="button" className="b--mat b--affirmative">
              view job list
            </button> */}
          {/* <button
              type="button"
              onClick={this.packageClick}
              className="b--mat b--affirmative"
            >
              Post Deploy Management
            </button> */}
        </div>
      </div>
      <div className="project-summary-metrics-container" />
    </div>
  );
};

ProjectSummary.propTypes = {
  project: PropTypes.object,
  history: PropTypes.object
};

ProjectSummary.defaultProps = {
  project: {},
  history: {}
};

export default withRouter(ProjectSummary);
