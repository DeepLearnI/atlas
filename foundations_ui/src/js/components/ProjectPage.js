import React, { Component } from 'react';
import Toolbar from './Toolbar';
import ProjectActions from '../actions/ProjectActions';
import ProjectHeader from './ProjectHeader';
import ProjectSummary from './ProjectSummary';

class ProjectPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoaded: false,
      projects: [],
    };
    this.getAllProjects = this.getAllProjects.bind(this);
  }

  componentDidMount() {
    this.getAllProjects();
  }

  async getAllProjects() {
    const apiProjects = await ProjectActions.getProjects();
    if (apiProjects != null) {
      this.setState({ projects: apiProjects, isLoaded: true });
    } else {
      this.setState({ projects: [], isLoaded: true });
    }
  }

  render() {
    const { isLoaded, projects } = this.state;
    let projectList;
    if (isLoaded) {
      if (projects.length === 0) {
        projectList = <p>No projects available</p>;
      } else {
        projectList = [];
        projects.forEach((project) => {
          const key = project.name.concat('-').concat(project.created_at);
          projectList.push(<ProjectSummary key={key} project={project} />);
        });
      }
    } else {
      projectList = <p>Loading projects</p>;
    }

    return (
      <div className="project-page-container">
        <Toolbar />
        <ProjectHeader />
        <div className="projects-body-container">
          {projectList}
        </div>
      </div>
    );
  }
}

export default ProjectPage;
