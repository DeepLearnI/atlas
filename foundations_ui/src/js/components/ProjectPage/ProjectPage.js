import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import Toolbar from '../common/Toolbar';
import ProjectHeader from './ProjectHeader';
import Loading from '../common/Loading';
import BaseActions from '../../actions/BaseActions';
import CommonActions from '../../actions/CommonActions';
import ProjectSummary from './ProjectSummary';
import CommonHeader from '../common/CommonHeader';
import CommonFooter from '../common/CommonFooter';

const ProjectPage = props => {
  const [isLoading, setIsLoading] = React.useState(false);
  const [projects, setProjects] = React.useState([]);

  const setTagsForProjects = fetchedProjects => {
    const newProjects = [];
    const promises = fetchedProjects.map(fetchedProject => {
      const newProject = fetchedProject;
      return BaseActions.getFromStaging(`projects/${fetchedProject.name}/job_listing`)
        .then(result => {
          if (result && result.jobs) {
            const tags = CommonActions.getTagsFromJob(result.jobs);
            newProject.tags = tags;
            return newProject;
          }
        });
    });

    return Promise.all(promises).then(result => {
      return result;
    });
  };

  const reload = () => {
    setIsLoading(true);

    BaseActions.getFromStaging('projects').then(result => {
      if (result != null) {
        result.sort((a, b) => {
          const dateA = new Date(a.created_at);
          const dateB = new Date(b.created_at);

          return dateB - dateA;
        });
        setTagsForProjects(result).then(updatedProjects => {
          setProjects(updatedProjects);
          setIsLoading(false);
        });
      }
    }).catch(() => {
      setIsLoading(false);
    });
  };

  React.useEffect(() => {
    reload();
  }, []);

  const renderProjects = () => {
    if (isLoading) {
      return <Loading loadingMessage="We are currently loading your projects" />;
    }
    if (projects.length === 0) {
      return <p>No projects available</p>;
    }
    return projects.map(project => {
      const newProject = project;
      const key = newProject.name.concat('-').concat(newProject.created_at);
      const formattedDate = moment(newProject.created_at)
        .format('YYYY-MM-DD HH:mm')
        .toString();
      newProject.created_at = formattedDate;
      return (
        <ProjectSummary
          key={key}
          project={newProject}
          selectProject={props.selectProject}
          changePage={props.changePage}
        />
      );
    });
  };

  return (
    <div>
      <CommonHeader isProject history={props.history} />
      <div className="project-page-container">
        <h1>Project Directory</h1>
        <div className="projects-body-container">{renderProjects()}</div>
      </div>
      <CommonFooter />
    </div>
  );
};


ProjectPage.propTypes = {
  selectProject: PropTypes.func,
  changePage: PropTypes.func,
  history: PropTypes.object,
};

ProjectPage.defaultProps = {
  selectProject: () => null,
  changePage: () => null,
  history: {},
};

export default ProjectPage;
