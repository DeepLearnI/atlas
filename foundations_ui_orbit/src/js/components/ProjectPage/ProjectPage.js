import React from 'react';
import PropTypes from 'prop-types';
import Toolbar from '../common/Toolbar';
import ProjectHeader from './ProjectHeader';
import Loading from '../common/Loading';
import { get } from '../../actions/BaseActions';
import ProjectSummary from './ProjectSummary';
import moment from 'moment';

class ProjectPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isLoading: false,
      projects: [],
      timerId: -1,
    };

    this.reload = this.reload.bind(this);
    this.startTimer = this.startTimer.bind(this);
    this.stopTimer = this.stopTimer.bind(this);
  }

  reload(showAnimation) {
    if (showAnimation === true) {
      this.setState({
        isLoading: true,
      });
    }

    get('projects').then(result => {
      if (result != null) {
        result.sort((a, b) => {
          const dateA = new Date(a.created_at);
          const dateB = new Date(b.created_at);

          return dateB - dateA;
        });
        this.setState({
          projects: result,
        });
      }
      this.setState({
        isLoading: false,
      });
    }).catch(() => {
      this.setState({
        isLoading: false,
      });
    });
  }

  startTimer() {
    const id = setInterval(() => {
      this.reload(false);
    }, 30000);
    this.setState({
      timerId: id,
    });
  }

  stopTimer() {
    const { timerId } = this.state;
    clearInterval(timerId);
  }

  componentDidMount() {
    this.reload(true);
    this.startTimer();
  }

  componentWillUnmount() {
    this.stopTimer();
  }

  renderProjects() {
    const { isLoading, projects } = this.state;
    const { selectProject, changePage } = this.props;
    if (isLoading) {
      return <Loading loadingMessage="We are currently loading your projects" />;
    }
    if (projects.length === 0) {
      return <p>No projects available</p>;
    }
    return projects.map(project => {
      const newProject = project;
      const key = newProject.name.concat('-').concat(newProject.created_at);
      const formattedDate = moment(newProject.created_at)
        .format('YYYY-MM-DD HH:mm')
        .toString();
      newProject.created_at = formattedDate;
      return (
        <ProjectSummary
          key={key}
          project={newProject}
          selectProject={selectProject}
          changePage={changePage}
        />
      );
    });
  }

  render() {
    const { projects } = this.state;
    return (
      <div className="project-page-container">
        <div className="header">
          <Toolbar />
          <ProjectHeader numProjects={projects.length} />
        </div>
        <div className="projects-body-container">{this.renderProjects()}</div>
      </div>
    );
  }
}

ProjectPage.propTypes = {
  selectProject: PropTypes.func,
  changePage: PropTypes.func,
};

ProjectPage.defaultProps = {
  selectProject: () => null,
  changePage: () => null,
};

export default ProjectPage;
