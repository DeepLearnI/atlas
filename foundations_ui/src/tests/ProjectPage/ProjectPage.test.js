import React from 'react';
import ReactDOM from 'react-dom';
import ProjectPage from '../../js/components/ProjectPage/ProjectPage';
import { shallow, mount } from 'enzyme';
import configureTests from '../setupTests';

configureTests();

it('Shallow Renders ProjectPage', () => {
  const wrapper = shallow(<ProjectPage/>);
  expect(wrapper).toMatchSnapshot();
});

it('Calls Get All Projects', async () => {
  const wrapper = mount(<ProjectPage/>); 
  const preState = wrapper.state();
  await wrapper.instance().getAllProjects();
  const postState = wrapper.state();
  expect(preState.isLoaded === false);
  expect(postState.isLoaded === true);
});

// Assumes your API has projects
it('Has at least One Project', async () => {
  const wrapper = mount(<ProjectPage/>); 
  const preState = wrapper.state();
  await wrapper.instance().getAllProjects();
  const postState = wrapper.state();
  expect(preState.projects.length === 0);
  expect(postState.projects.length > 0);
});
