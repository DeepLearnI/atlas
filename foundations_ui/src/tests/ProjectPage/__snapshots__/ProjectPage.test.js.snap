// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders ProjectPage 1`] = `
<div
  className="project-page-container"
>
  <div
    className="header"
  >
    <Toolbar />
    <ProjectHeader
      numProjects={0}
    />
  </div>
  <div
    className="projects-body-container"
  >
    <Loading
      loadingMessage="We are currently loading your projects"
    />
  </div>
</div>
`;
